harness/determined/pytorch/samplers.py

```diff
@@ -0,0 +1,361 @@
+"""
+Guidelines for Reproducible Datasets.
+
+.. note::
+
+   If you are using a MapDataset and not an IterableDataset, make your life easy and use the
+   context.TrainingDataLoader() and context.ValidationDataLoader(), which are a drop-in replacement
+   for pytorch's built-in DataLoader, and Determined will handle all of these details.
```

Comment:
add link to trainingdataloader docs

Reply:
also can we add a link of why you would use this approach: i.e. why do i need to use iterable dataset here and not the easy dropin

Reply:
I actually think I dislike the `context.TrainingDataLoader` and `context.ValidationDataLoader()`.  I think I would feel more comfortable adding *only* the low-level API in this PR, and leave the high-level API as just the `det.pytorch.DataLoader()` (which is what we currently have)

Reply:
> also can we add a link of why you would use this approach: i.e. why do i need to use iterable dataset here and not the easy dropin

Hm?  How do I know why you need to use an iterable dataset?  I feel like that's the user's concern